jaggery-modules/taxonomy/module/scripts/taxonomy/taxonomy.js

```diff
@@ -45,8 +47,21 @@ var GovernanceAPITree = org.wso2.carbon.governance.api.util.TaxonomyTreeAPI;
      * @return {Object}  Categories list as paths, which admin defined.
      */
     taxonomy.getTaxa = function () {
-        return JSON.parse(GovernanceCommon.getPathCategories());
+        if (!session.get(TAXA)) {
+            return putInSession();
```

Comment:
Thank you Sameera for suggestion. This entire code block has been refactored with multiple taxonomy.